src/kit.ts

```diff
@@ -292,11 +292,33 @@ export async function kitIfCompiler(bin: string, pr?: ProgressReporter): Promise
         if (version === null) {
             return null;
         }
-        const gxx_fname = fname.replace(/gcc/, 'g++');
-        const gxx_bin = path.join(path.dirname(bin), gxx_fname);
-        const gccCompilers: { [lang: string]: string } = { C: bin };
-        if (await fs.exists(gxx_bin)) {
-            gccCompilers.CXX = gxx_bin;
+        const gccCompilers: { [lang: string]: string } = { };
+        const gxx_fname1 = fname.replace(/gcc/, 'g++');
+        const gxx_bin1 = path.join(path.dirname(bin), gxx_fname1);
+        if (await fs.exists(gxx_bin1)) {
+            // Names like x86_64-pc-linux-gnu-gcc-11.1.0
+            gccCompilers.C = bin;
+            // Names like x86_64-pc-linux-gnu-g++-11.1.0
+            gccCompilers.CXX = gxx_bin1;
+        } else {
+            const fname2 = fname.replace(/gcc(-\d+(\.\d+(\.\d+)?)?)/, 'gcc');
+            const bin2 = path.join(path.dirname(bin), fname2);
+            const gxx_fname2 = fname2.replace(/gcc/, 'g++');
+            const gxx_bin2 = path.join(path.dirname(bin), gxx_fname2);
+            // Ensure the version is match
+            const version2 = await getCompilerVersion('GCC', bin2, pr);
```

Comment:
I think the part I struggle with here is that `bin2` === `bin1` if no substitution is made, so `version_is_match` will be true because it's the same compiler.  Looking at the code, it seems like the correct result is produced because `gxx_bin2` is the same as `gxx_bin1` (it's basically the same as the code we're replacing); It's just not immediately clear from the code that that is what's happening.  Could you add a comment or two here explaining this case?

Reply:
Sure.
For the kits like:

```
x86_64-pc-linux-gnu-gcc-11.1.0
x86_64-pc-linux-gnu-gcc
x86_64-pc-linux-gnu-g++
```
It will find the `x86_64-pc-linux-gnu-gcc-11.1.0` first.

As it can not find `x86_64-pc-linux-gnu-gcc-11.1.0` which is `gxx_bin1`, it will goes to check the bin like `x86_64-pc-linux-gnu-gcc`. 

If exists and the version is match, it will use the `x86_64-pc-linux-gnu-gcc` instead.

And it will detect the `x86_64-pc-linux-gnu-g++`, if exists set, otherwise not.

---

For the kits like:

(which is not passed before)

```
x86_64-pc-linux-gnu-gcc
```

It will find the `x86_64-pc-linux-gnu-gcc` first.

As the `x86_64-pc-linux-gnu-g++` does not exist, it will goes to `bin2`'s check.

`bin2` is the same as `bin` which are all `x86_64-pc-linux-gnu-gcc`, hence, only the c compiler will be set.

---

For the kits like:

```
x86_64-pc-linux-gnu-gcc-11.1.0
```

It will find the `x86_64-pc-linux-gnu-gcc-11.1.0` first.

As the `x86_64-pc-linux-gnu-g++-11.1.0` does not exist, it will goes to `bin2`'s check.

`bin2` is `x86_64-pc-linux-gnu-gcc`, which does not exist. It will fall back to `bin` and then only the c compiler will be set.

Reply:
Sorry if I was unclear.  I was asking if you could put a brief comment in the code about this.  Something like "if bin2 === bin1 because the regex did not make a replacement, then version_match will be true, but no C++ compiler will be found, so we will correctly skip setting the CXX compiler."

Thank you.

Reply:
Sorry for my misunderstanding. I will add comments on this later.

Reply:
Thank you.